python/sglang/srt/server_args.py

```diff
@@ -412,6 +413,12 @@ def add_cli_args(parser: argparse.ArgumentParser):
             default=ServerArgs.download_dir,
             help="Model download directory.",
         )
+        parser.add_argument(
+            "--base-gpu-id",
```

Comment:
can we use CUDA_VISIBLE_DEVICES to achieve this?

Reply:
https://github.com/sgl-project/sglang/pull/2114/files#diff-38af500b72b9f4e46bd0beafdbad7f7cfdbbe82a6dfd3284b18d7eb4948adb8eR35

Here is my current impl of the new `--dp-size`, which launches multiple TP server before the router starts. With base gpu id, the implementation is quite clean. Pretty much a decoupled version of the existing dp controller, where we also have a based gpu id.

 I have tried `CUDA_VISIBLE_DEVICES`, but it is not set correctly in the subprocess for some reasons. Another reason is that base-gpu-id can work with AMD too and `CUDA_VISIBLE_DEVICES` is only for NVIDIA.
 
 Any concern of adding this?

Reply:
ok, we can use this.

Can you also fix this for `--dp`?

Reply:
Do you mean add `serving_args.base_gpu_id +` to https://github.com/sgl-project/sglang/blob/f6f713797bcbc63d225136d66deaa00495cdedfe/python/sglang/srt/managers/data_parallel_controller.py#L159?

Reply:
After setting `CUDA_VISIBLE_DEVICES`
e.g.
```
export CUDA_VISIBLE_DEVICES=4,6
```
Then the program can only see the 0 and 1. This IDs have been automatically rearranged. The 0 and 1 here actually refer to 4 and 6.